docs/developer-docs/developer-tools/off-chain/agents/rust-agent.mdx

```diff
@@ -24,6 +24,104 @@ To add the `ic-agent` crate as a dependency in your project, use the command:
 cargo add ic-agent
 ```
 
+## Initializing the agent
+
+Before using the agent in your project, it must be initialized using the `Agent::builder()` function. This can be done within your Rust code, or in a separate `agent.rs` file that can be imported into other Rust files. Here is an example of how to initialize the Rust agent:
+
+```rust
+use anyhow::Result;
+use ic_agent::{agent::http_transport::reqwest_transport::ReqwestHttpReplicaV2Transport, Agent};
+
+pub async fn create_agent(url: &str) -> Result<Agent> {
+    let transport = ReqwestHttpReplicaV2Transport::create(url)?;
+
+    let agent = Agent::builder().with_transport(transport).build()?;
+    agent.fetch_root_key().await?;
+
+    Ok(agent)
+}
+```
+
+
+## Authentication
+
+The Rust agent's `Identity` object provides signatures that can be used for HTTP requests or identity delegations. It represents the principal ID of the sender.
+
+Identities can be created when the agent is initialized using the `.with_identity(create_identity())` method. Each agent has a single `Identity` object that can contain multiple identity values.
+
+```rust
+async fn create_a_canister() -> Result<Principal, Box<dyn std::error::Error>> {
+  let agent = Agent::builder()
+    .with_url(URL)
+    .with_identity(create_identity())
+    .build()?;
+```
+
+
+## Making calls
+
+The Rust agent can be used to make calls to other canisters. To make a call to a canister, use the `agent.update` method for an update call, or `agent.query` for a query call. Then, pass in the canister's ID and the method of the canister you'd like to call.
+
+For example, to call the management canister's `provisional_create_canister_with_cycles` method, use the following code:
+
+```rust
+  let effective_canister_id = Principal::from_text("rwlgt-iiaaa-aaaaa-aaaaa-cai").unwrap();
+  let response = agent.update(&management_canister_id, "provisional_create_canister_with_cycles")
+    .with_effective_canister_id(effective_canister_id)
+    .with_arg(Encode!(&Argument { amount: None})?)
+    .call_and_wait()
+    .await?;
+
+  let result = Decode!(response.as_slice(), CreateCanisterResult)?;
+  let canister_id: Principal = result.canister_id;
+  Ok(canister_id)
+```
+
+## Asset certification
```

Comment:
`ic-http-certification` doesn't have anything to do with the Rust agent. It is used in canisters to implement their end of the HTTP gateway, and its certifications are verified by the boundary node, not the agent.

We probably do want a response verification example of *some* kind, which would go over how to use `HashTree` when you're using a canister that makes use of certified queries, but I don't know if there's any written-up example you can pull from. [This dfx function](https://github.com/dfinity/sdk/blob/f0baed19e47e68bc2752aa441a1c1164c4e9bca3/src/dfx/src/lib/operations/ledger.rs#L50) could be a starting point but I remember when writing it that documentation on how to use the moving parts was very scattered.

The other thing you might put in an 'asset certification' section is the client API for the asset canister itself, which is found [here](https://github.com/dfinity/sdk/tree/master/src/canisters/frontend/ic-asset) and similarly does not have good pre-existing examples or documentation.

Reply:
Revised per feedback